README.md

```diff
@@ -231,172 +257,278 @@ curl -s -i https://api.travis-ci.org/auth/github \
+System packages are marked with `MASON_SYSTEM_PACKAGE=true`. We're also first using `pkg-config` to check whether the library is present at all. The `mason_system_version` function creates a small executable which outputs the actual version. It is the only thing that is cached in the installation directory.
+
+We have to override the `mason_cflags` and `mason_ldflags` commands since the regular commands return flags for static libraries, but in the case of system packages, we want to dynamically link against the package.
+
+## Releasing a package
+
+Here is an example workflow to help get you started:
+
+1. Create an annotated tag in git for the code you want to package.
+
+    Annotated tags can be stored, checksummed, signed and verified with GNU Privacy Guard (GPG) in Github. To create an annotated tag specify `-a` when running the `tag` command, for example:
+
+    `git tag -a v0.1.0 -m "version 0.1.0"`
+
+2. Share your new tag.
+
+    You have to explicitly push your new tag to a shared Github server. This is the location we will share with Mason when specifying where to download the code to be packaged. Using our example above we would run:
+
+    `git push origin v0.1.0`
+
+    (Or you can push all tags: `git push --tags`.)
+
+3. Create a package.
+
+    We recommend working in a new branch before creating a pacakge. For example if you want to call your new package `my_new_package` version `0.1.0` you could create and checkout a branch like this:
+
+    `git checkout -b my_new_package-0.1.0`
+
+    Now follow the instructions in the [Getting started](#getting-started) section for creating a new package.
+
+4. Test your package.
+
+    Even though we will eventually build the package using Travis, it's a good idea to build locally to check for errors.
+
+     `./mason build my_new_package 0.1.0`
+
+5. Push changes to remote.
+
+    Once you can build, push your changes up to Github remote so that Travis will know what to build and publish in the next step.
+
+    `git push origin my_new_package-0.1.0`
+
+6. Build and Publish your package.
+
+    Use Mason's `trigger` command to tell Travis to build, test, and publish your new package to the S3 bucket specified in Mason. By default this S3 bucket is set to a Mapbox-managed S3 bucket called `mason-binaries`.
+
+    `./mason trigger my_new_package 0.1.0`
+
+7. Check your S3 bucket to see that your package exists.
 
-#### Script structure
+    The location of your package depends on how you configured Mason (see [Configuration](#configuration)) and the type of package you published. For example if you published your package to the default S3 bucket (`mason-binaries`) and it's a header-only package, you can find your package here:
 
-This `script.sh` is structured like:
+    `aws s3 ls s3://mason-binaries/headers/my_new_package/0.1.0.tar.gz`
+
+## Using a package
+
+1. First make sure you followed the [Mason installation](#installation) instructions.
+
```

Comment:
@allieoop I would say something like: 

Mason has two clients for installing packages - 
1) The Mason cli tool - you can follow the install instructions in the Mason docs <link> The cli install client is part of the Mason project itself. 
2) Mason-js - you can follow its docs here: https://github.com/mapbox/mason-js. Mason-js is a separate node JS client. 

There are examples of projects using the cli tool and mason-js. 
* [hpp-skell](https://github.com/mapbox/hpp-skel) uses the cli tool. This install client requires you to update your Mason version in `scripts/setup.sh`
 * [node-cpp-skel](https://github.com/mapbox/node-cpp-skel) uses mason-js, a JS client that pulls packages directly from S3.

The install command syntax will differ depending on the client you use.